woodwork/table_accessor.py

```diff
@@ -251,3 +264,17 @@ def _update_column_dtype(series, logical_type, name):
                 f'logical type {logical_type}.'
             raise TypeError(error_msg)
     return series
+
+
+def _is_valid_schema(dataframe, schema):
+    if set(dataframe.columns) != set(schema.columns.keys()):
+        return False
+    for name in dataframe.columns:
+        if str(dataframe[name].dtype) != schema.logical_types[name].pandas_dtype:
+            return False
+    if schema.index is not None:
```

Comment:
Do you need to check the time index?

Reply:
I don't think so?

The things to check would be whether the data for a possible `schema.index` is an acceptable Datetime or Numeric Datetime, but since the time_index is set in the Schema, we know it has to be a `Datetime` logical type or have the `numeric` tag, and since we also check that the dtype of each series matches its Logical Type, I think that means that the column should only contain valid time index data. 

But I definitely could be missing something to check for the time index here.

Reply:
Gotcha.